## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 496778`
Commit: `d243d3f1de9881a25c8f83821420310bad181974`

---
### `aarch64-darwin`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>python313Packages.langchain-google-genai</li>
    <li>python313Packages.langchain-google-genai.dist</li>
    <li>python313Packages.langchain-huggingface</li>
    <li>python313Packages.langchain-huggingface.dist</li>
    <li>python314Packages.langchain-google-genai</li>
    <li>python314Packages.langchain-google-genai.dist</li>
    <li>python314Packages.langchain-huggingface</li>
    <li>python314Packages.langchain-huggingface.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 146 packages built:</summary>
  <ul>
    <li>langgraph-cli (python313Packages.langgraph-cli)</li>
    <li>langgraph-cli.dist (python313Packages.langgraph-cli.dist)</li>
    <li>mlflow-server</li>
    <li>mlflow-server.dist</li>
    <li>python313Packages.databricks-sdk</li>
    <li>python313Packages.databricks-sdk.dist</li>
    <li>python313Packages.great-tables</li>
    <li>python313Packages.great-tables.dist</li>
    <li>python313Packages.langchain</li>
    <li>python313Packages.langchain-anthropic</li>
    <li>python313Packages.langchain-anthropic.dist</li>
    <li>python313Packages.langchain-aws</li>
    <li>python313Packages.langchain-aws.dist</li>
    <li>python313Packages.langchain-azure-dynamic-sessions</li>
    <li>python313Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python313Packages.langchain-chroma</li>
    <li>python313Packages.langchain-chroma.dist</li>
    <li>python313Packages.langchain-classic</li>
    <li>python313Packages.langchain-classic.dist</li>
    <li>python313Packages.langchain-community</li>
    <li>python313Packages.langchain-community.dist</li>
    <li>python313Packages.langchain-core</li>
    <li>python313Packages.langchain-core.dist</li>
    <li>python313Packages.langchain-deepseek</li>
    <li>python313Packages.langchain-deepseek.dist</li>
    <li>python313Packages.langchain-experimental</li>
    <li>python313Packages.langchain-experimental.dist</li>
    <li>python313Packages.langchain-fireworks</li>
    <li>python313Packages.langchain-fireworks.dist</li>
    <li>python313Packages.langchain-groq</li>
    <li>python313Packages.langchain-groq.dist</li>
    <li>python313Packages.langchain-mistralai</li>
    <li>python313Packages.langchain-mistralai.dist</li>
    <li>python313Packages.langchain-mongodb</li>
    <li>python313Packages.langchain-mongodb.dist</li>
    <li>python313Packages.langchain-ollama</li>
    <li>python313Packages.langchain-ollama.dist</li>
    <li>python313Packages.langchain-openai</li>
    <li>python313Packages.langchain-openai.dist</li>
    <li>python313Packages.langchain-perplexity</li>
    <li>python313Packages.langchain-perplexity.dist</li>
    <li>python313Packages.langchain-tests</li>
    <li>python313Packages.langchain-tests.dist</li>
    <li>python313Packages.langchain-text-splitters</li>
    <li>python313Packages.langchain-text-splitters.dist</li>
    <li>python313Packages.langchain-xai</li>
    <li>python313Packages.langchain-xai.dist</li>
    <li>python313Packages.langchain.dist</li>
    <li>python313Packages.langgraph</li>
    <li>python313Packages.langgraph-checkpoint</li>
    <li>python313Packages.langgraph-checkpoint-mongodb</li>
    <li>python313Packages.langgraph-checkpoint-mongodb.dist</li>
    <li>python313Packages.langgraph-checkpoint-postgres</li>
    <li>python313Packages.langgraph-checkpoint-postgres.dist</li>
    <li>python313Packages.langgraph-checkpoint-sqlite</li>
    <li>python313Packages.langgraph-checkpoint-sqlite.dist</li>
    <li>python313Packages.langgraph-checkpoint.dist</li>
    <li>python313Packages.langgraph-prebuilt</li>
    <li>python313Packages.langgraph-prebuilt.dist</li>
    <li>python313Packages.langgraph-runtime-inmem</li>
    <li>python313Packages.langgraph-runtime-inmem.dist</li>
    <li>python313Packages.langgraph-store-mongodb</li>
    <li>python313Packages.langgraph-store-mongodb.dist</li>
    <li>python313Packages.langgraph.dist</li>
    <li>python313Packages.langsmith</li>
    <li>python313Packages.langsmith.dist</li>
    <li>python313Packages.mlflow</li>
    <li>python313Packages.mlflow.dist</li>
    <li>python313Packages.mmcv</li>
    <li>python313Packages.mmcv.dist</li>
    <li>python313Packages.mmengine</li>
    <li>python313Packages.mmengine.dist</li>
    <li>python313Packages.sagemaker-mlflow</li>
    <li>python313Packages.sagemaker-mlflow.dist</li>
    <li>python313Packages.shiny</li>
    <li>python313Packages.shiny.dist</li>
    <li>python313Packages.torchtune</li>
    <li>python313Packages.torchtune.dist</li>
    <li>python314Packages.databricks-sdk</li>
    <li>python314Packages.databricks-sdk.dist</li>
    <li>python314Packages.great-tables</li>
    <li>python314Packages.great-tables.dist</li>
    <li>python314Packages.langchain</li>
    <li>python314Packages.langchain-anthropic</li>
    <li>python314Packages.langchain-anthropic.dist</li>
    <li>python314Packages.langchain-aws</li>
    <li>python314Packages.langchain-aws.dist</li>
    <li>python314Packages.langchain-azure-dynamic-sessions</li>
    <li>python314Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python314Packages.langchain-classic</li>
    <li>python314Packages.langchain-classic.dist</li>
    <li>python314Packages.langchain-community</li>
    <li>python314Packages.langchain-community.dist</li>
    <li>python314Packages.langchain-core</li>
    <li>python314Packages.langchain-core.dist</li>
    <li>python314Packages.langchain-deepseek</li>
    <li>python314Packages.langchain-deepseek.dist</li>
    <li>python314Packages.langchain-experimental</li>
    <li>python314Packages.langchain-experimental.dist</li>
    <li>python314Packages.langchain-fireworks</li>
    <li>python314Packages.langchain-fireworks.dist</li>
    <li>python314Packages.langchain-groq</li>
    <li>python314Packages.langchain-groq.dist</li>
    <li>python314Packages.langchain-mistralai</li>
    <li>python314Packages.langchain-mistralai.dist</li>
    <li>python314Packages.langchain-mongodb</li>
    <li>python314Packages.langchain-mongodb.dist</li>
    <li>python314Packages.langchain-ollama</li>
    <li>python314Packages.langchain-ollama.dist</li>
    <li>python314Packages.langchain-openai</li>
    <li>python314Packages.langchain-openai.dist</li>
    <li>python314Packages.langchain-perplexity</li>
    <li>python314Packages.langchain-perplexity.dist</li>
    <li>python314Packages.langchain-tests</li>
    <li>python314Packages.langchain-tests.dist</li>
    <li>python314Packages.langchain-text-splitters</li>
    <li>python314Packages.langchain-text-splitters.dist</li>
    <li>python314Packages.langchain-xai</li>
    <li>python314Packages.langchain-xai.dist</li>
    <li>python314Packages.langchain.dist</li>
    <li>python314Packages.langgraph</li>
    <li>python314Packages.langgraph-checkpoint</li>
    <li>python314Packages.langgraph-checkpoint-mongodb</li>
    <li>python314Packages.langgraph-checkpoint-mongodb.dist</li>
    <li>python314Packages.langgraph-checkpoint-postgres</li>
    <li>python314Packages.langgraph-checkpoint-postgres.dist</li>
    <li>python314Packages.langgraph-checkpoint-sqlite</li>
    <li>python314Packages.langgraph-checkpoint-sqlite.dist</li>
    <li>python314Packages.langgraph-checkpoint.dist</li>
    <li>python314Packages.langgraph-cli</li>
    <li>python314Packages.langgraph-cli.dist</li>
    <li>python314Packages.langgraph-prebuilt</li>
    <li>python314Packages.langgraph-prebuilt.dist</li>
    <li>python314Packages.langgraph-runtime-inmem</li>
    <li>python314Packages.langgraph-runtime-inmem.dist</li>
    <li>python314Packages.langgraph-store-mongodb</li>
    <li>python314Packages.langgraph-store-mongodb.dist</li>
    <li>python314Packages.langgraph.dist</li>
    <li>python314Packages.langsmith</li>
    <li>python314Packages.langsmith.dist</li>
    <li>python314Packages.mlflow</li>
    <li>python314Packages.mlflow.dist</li>
    <li>python314Packages.sagemaker-mlflow</li>
    <li>python314Packages.sagemaker-mlflow.dist</li>
    <li>python314Packages.shiny</li>
    <li>python314Packages.shiny.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python313Packages.langchain-google-genai</summary>
<pre>E             Omitting 7 identical items, use -vv to show
E             Differing items:
E             {&#x27;grounding_chunks&#x27;: [{&#x27;image&#x27;: None, &#x27;maps&#x27;: None, &#x27;retrieved_context&#x27;: None, &#x27;web&#x27;: {&#x27;domain&#x27;: None, &#x27;title&#x27;: ...
E             
E             ...Full output truncated (4 lines hidden), use &#x27;-vv&#x27; to show

tests/unit_tests/test_chat_models.py:1477: AssertionError

----------------------------------------------- benchmark: 1 tests -----------------------------------------------
Name (time in ms)          Min       Max      Mean  StdDev    Median     IQR  Outliers     OPS  Rounds  Iterations
------------------------------------------------------------------------------------------------------------------
test_init_time        182.8217  192.2486  187.7008  3.7742  187.3541  7.2010       3;0  5.3276       6           1
------------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
=========================== short test summary info ============================
FAILED tests/unit_tests/test_chat_models.py::test_response_to_result_grounding_metadata[raw_response0-expected_grounding_metadata0] - AssertionError: assert {&#x27;google_maps...s&#x27;: None, ...} == {&#x27;google_maps...a&#x27;...
================= 1 failed, 248 passed, 2 deselected in 5.72s ==================</pre>
</details>
<details>
<summary>python313Packages.langchain-huggingface</summary>
<pre>no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Successfully built langchain_huggingface-1.2.0-py3-none-any.whl
Finished creating a wheel...
/nix/var/nix/b/6jq0mhg0zdz8ywzd0nqjpgp8a9/source/libs/partners/huggingface/dist /nix/var/nix/b/6jq0mhg0zdz8ywzd0nqjpgp8a9/source/libs/partners/huggingface
Unpacking to: unpacked/langchain_huggingface-1.2.0...OK
Repacking wheel as ./langchain_huggingface-1.2.0-py3-none-any.whl...OK
/nix/var/nix/b/6jq0mhg0zdz8ywzd0nqjpgp8a9/source/libs/partners/huggingface
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonRuntimeDepsCheckHook&quot; }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for langchain_huggingface-1.2.0-py3-none-any.whl
  - huggingface-hub&lt;1.0.0,&gt;=0.33.4 not satisfied by version 1.4.1</pre>
</details>
<details>
<summary>python314Packages.langchain-google-genai</summary>
<pre>    VersionedUnionType = Union[builtin_types.UnionType, _UnionGenericAlias]

../../../../../../../store/23y0fn08mfl7zz88fa68ljl1ys2l32sb-python3.14-langchain-core-1.2.12/lib/python3.14/site-packages/langchain_core/_api/deprecation.py:25
  /nix/store/23y0fn08mfl7zz88fa68ljl1ys2l32sb-python3.14-langchain-core-1.2.12/lib/python3.14/site-packages/langchain_core/_api/deprecation.py:25: UserWarning: Core Pydantic V1 functionality isn&#x27;t compatible with Python 3.14 or greater.
    from pydantic.v1.fields import FieldInfo as FieldInfoV1

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

----------------------------------------------- benchmark: 1 tests -----------------------------------------------
Name (time in ms)          Min       Max      Mean  StdDev    Median     IQR  Outliers     OPS  Rounds  Iterations
------------------------------------------------------------------------------------------------------------------
test_init_time        176.4359  190.7505  184.1301  4.9294  184.2933  5.6162       2;0  5.4309       6           1
------------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
=========================== short test summary info ============================
FAILED tests/unit_tests/test_chat_models.py::test_response_to_result_grounding_metadata[raw_response0-expected_grounding_metadata0] - AssertionError: assert {&#x27;google_maps...s&#x27;: None, ...} == {&#x27;google_maps...a&#x27;...
=========== 1 failed, 248 passed, 2 deselected, 2 warnings in 5.82s ============</pre>
</details>
<details>
<summary>python314Packages.langchain-huggingface</summary>
<pre>no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Successfully built langchain_huggingface-1.2.0-py3-none-any.whl
Finished creating a wheel...
/nix/var/nix/b/0sy2gvaa4jqdl6dmp834ra6rn3/source/libs/partners/huggingface/dist /nix/var/nix/b/0sy2gvaa4jqdl6dmp834ra6rn3/source/libs/partners/huggingface
Unpacking to: unpacked/langchain_huggingface-1.2.0...OK
Repacking wheel as ./langchain_huggingface-1.2.0-py3-none-any.whl...OK
/nix/var/nix/b/0sy2gvaa4jqdl6dmp834ra6rn3/source/libs/partners/huggingface
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonRuntimeDepsCheckHook&quot; }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for langchain_huggingface-1.2.0-py3-none-any.whl
  - huggingface-hub&lt;1.0.0,&gt;=0.33.4 not satisfied by version 1.4.1</pre>
</details>
</details>

--- a/pkgs/development/python-modules/langsmith/default.nix
+++ b/pkgs/development/python-modules/langsmith/default.nix
@@ -1,6 +1,5 @@
 {
   lib,
-  stdenv,
   buildPythonPackage,
   fetchFromGitHub,
 
@@ -14,6 +13,7 @@
   requests,
   requests-toolbelt,
   uuid-utils,
+  xxhash,
   zstandard,
 
   # tests
@@ -23,6 +23,7 @@
   multipart,
   opentelemetry-sdk,
   pytest-asyncio,
+  pytest-httpx,
   pytest-socket,
   pytest-vcr,
   pytestCheckHook,
@@ -30,14 +31,14 @@
 
 buildPythonPackage rec {
   pname = "langsmith";
-  version = "0.6.4";
+  version = "0.7.12";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "langchain-ai";
     repo = "langsmith-sdk";
     tag = "v${version}";
-    hash = "sha256-325A2kEx2UrykxVRzp6WQCPrg92Vy+6R1CfgnCLV2V8=";
+    hash = "sha256-5BVQYlp/RMFN1Ja75gz9iFneqkSuoDpqXilHbyEAPdA=";
   };
 
   sourceRoot = "${src.name}/python";
@@ -53,6 +54,7 @@ buildPythonPackage rec {
     requests
     requests-toolbelt
     uuid-utils
+    xxhash
     zstandard
   ];
 
@@ -60,9 +62,10 @@ buildPythonPackage rec {
     anthropic
     attrs
     dataclasses-json
+    multipart
     opentelemetry-sdk
     pytest-asyncio
-    multipart
+    pytest-httpx
     pytest-socket
     pytest-vcr
     pytestCheckHook
@@ -93,6 +96,8 @@ buildPythonPackage rec {
     # flaky time comparisons
     # https://github.com/langchain-ai/langsmith-sdk/issues/2245
     "tests/unit_tests/test_uuid_v7.py"
+    # google-adk isn't packaged (and has an enormous number of dependencies)
+    "tests/unit_tests/wrappers/test_google_adk.py"
   ];
 
   pythonImportsCheck = [ "langsmith" ];

--- a/pkgs/development/python-modules/langsmith/default.nix
+++ b/pkgs/development/python-modules/langsmith/default.nix
@@ -29,7 +29,7 @@
   pytestCheckHook,
 }:
 
-buildPythonPackage rec {
+buildPythonPackage (finalAttrs: {
   pname = "langsmith";
   version = "0.7.12";
   pyproject = true;
@@ -37,11 +37,11 @@ buildPythonPackage rec {
   src = fetchFromGitHub {
     owner = "langchain-ai";
     repo = "langsmith-sdk";
-    tag = "v${version}";
+    tag = "v${finalAttrs.version}";
     hash = "sha256-5BVQYlp/RMFN1Ja75gz9iFneqkSuoDpqXilHbyEAPdA=";
   };
 
-  sourceRoot = "${src.name}/python";
+  sourceRoot = "${finalAttrs.src.name}/python";
 
   pythonRelaxDeps = [ "orjson" ];
 
@@ -107,7 +107,7 @@ buildPythonPackage rec {
   meta = {
     description = "Client library to connect to the LangSmith LLM Tracing and Evaluation Platform";
     homepage = "https://github.com/langchain-ai/langsmith-sdk";
-    changelog = "https://github.com/langchain-ai/langsmith-sdk/releases/tag/${src.tag}";
+    changelog = "https://github.com/langchain-ai/langsmith-sdk/releases/tag/${finalAttrs.src.tag}";
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [
       natsukium
@@ -115,4 +115,4 @@ buildPythonPackage rec {
     ];
     mainProgram = "langsmith";
   };
-}
+})

